packages/kbn-unified-doc-viewer/src/services/doc_views_registry.ts

```diff
@@ -18,24 +16,49 @@ export enum ElasticRequestState {
 }
 
 export class DocViewsRegistry {
-  private docViews: DocView[] = [];
+  private docViews: Map<string, DocView>;
+
+  constructor(initialValue?: DocViewsRegistry | DocView[]) {
+    if (initialValue instanceof DocViewsRegistry) {
+      this.docViews = new Map(initialValue.docViews);
+    } else if (Array.isArray(initialValue)) {
+      this.docViews = new Map(initialValue.map((docView) => [docView.id, docView]));
+    } else {
+      this.docViews = new Map();
+    }
+  }
 
-  /**
-   * Extends and adds the given doc view to the registry array
-   */
-  addDocView(docViewRaw: DocViewInput | DocViewInputFn) {
+  getAll() {
+    return [...this.docViews.values()];
+  }
+
+  add(docViewRaw: DocView | DocViewFactory) {
     const docView = typeof docViewRaw === 'function' ? docViewRaw() : docViewRaw;
-    this.docViews.push({
-      ...docView,
-      shouldShow: docView.shouldShow ?? (() => true),
-    });
+
+    if (this.docViews.has(docView.id)) {
+      throw new Error(
+        `DocViewsRegistry#add: a DocView is already registered with id "${docView.id}".`
+      );
+    }
+
+    this.docViews.set(docView.id, docView);
```

Comment:
Question: What about the use case where the client want to set a new DataView with an order which already exists ? In that we will have 2 dataViews lets say with the same order. Is this fine ?